x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -167,202 +124,159 @@ public void testNotStalledIfAllShardsHaveACopyOnAnotherNode() throws Exception {
                 .put(UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), 0) // Disable "normal" delayed allocation
         ).get();
         ensureGreen(indexName);
-        indexRandomData();
+        indexRandomData(indexName);
 
         String nodeToStopId = findIdOfNodeWithPrimaryShard(indexName);
-        PutShutdownNodeAction.Request putShutdownRequest = new PutShutdownNodeAction.Request(
-            nodeToStopId,
-            SingleNodeShutdownMetadata.Type.REMOVE,
-            this.getTestName(),
-            null,
-            null
-        );
-        AcknowledgedResponse putShutdownResponse = client().execute(PutShutdownNodeAction.INSTANCE, putShutdownRequest).get();
-        assertTrue(putShutdownResponse.isAcknowledged());
-        assertBusy(() -> {
-            GetShutdownStatusAction.Response getResp = client().execute(
-                GetShutdownStatusAction.INSTANCE,
-                new GetShutdownStatusAction.Request(nodeToStopId)
-            ).get();
 
-            assertThat(getResp.getShutdownStatuses().get(0).migrationStatus().getStatus(), equalTo(COMPLETE));
-        });
+        putNodeShutdown(nodeToStopId, SingleNodeShutdownMetadata.Type.REMOVE, null);
+        assertBusy(() -> assertNodeShutdownStatus(nodeToStopId, COMPLETE));
     }
 
     public void testNodeReplacementOnlyAllowsShardsFromReplacedNode() throws Exception {
-        String nodeA = internalCluster().startNode(Settings.builder().put("node.name", "node-a"));
-        Settings.Builder nodeASettings = Settings.builder().put("index.number_of_shards", 3).put("index.number_of_replicas", 1);
-        createIndex("myindex", nodeASettings.build());
-        final String nodeAId = getNodeId(nodeA);
-        final String nodeB = "node_t1"; // TODO: fix this to so it's actually overrideable
-
-        // Mark the nodeA as being replaced
-        PutShutdownNodeAction.Request putShutdownRequest = new PutShutdownNodeAction.Request(
-            nodeAId,
-            SingleNodeShutdownMetadata.Type.REPLACE,
-            this.getTestName(),
-            null,
-            nodeB
+        String node0 = internalCluster().startNode();// node_t0
+        String node0Id = getNodeId(node0);
+        String node1 = "node_t1";
+        String node2 = "node_t2";
```

Comment:
It's a bit yucky that we hard-code the node naming convention like this, although I see why we need to know the node names before they start up. Could we at least assert that the nodes have the right names when they start up?

Nothing to do with this change obviously, but still while we're here it would be nice to clear this up.